dp-cluster-setup-utility.py

```diff
@@ -1213,67 +1616,216 @@ def satisfied(self):
+class BaseOperationsManager(object):
+  @staticmethod
+  def get_instance():
+    pass
+  @staticmethod
+  def pre_reqs_satisfied(cluster_instance):
+    pass
+  
+  @staticmethod
+  def get_topology_util(cluster_instance, dp_dependencies):
+    pass
+  
+  @staticmethod
+  def get_knox_details(cluster_instance):
+    pass
 
-  role_names = map(lambda each: each.service_name, merged_dependencies)
+  @staticmethod
+  def get_topologies_to_deploy(dp, cluster_instance,topology_util):
+    pass
+  
+  @staticmethod
+  def check_cluster_for_dp(dp,knox):
+    pass
+  
+  @staticmethod
+  def register_cluster_with_dp(dp, cluster_instance,knox,user):
+    pass
+"""
+  AmbariOperationsManager : control Operation for ambari based clusters
 
+"""
 
-  topology_util = TopologyUtil(ambari, role_names)
+class AmbariOperationsManager(BaseOperationsManager):
+  
+  @staticmethod
+  def get_instance():
+    print "\nTell me about this cluster's Ambari Instance"
+    return Ambari(user.url_input('Ambari URL', 'ambari.url'), user.credential_input('Ambari admin', 'ambari.admin'))
+  
+  @staticmethod
+  def pre_reqs_satisfied(ambari):
+    return AmbariPrerequisites(ambari).satisfied()
+  
+  @staticmethod
+  def get_topology_util(ambari, dp):
+    merged_dependencies = dp.dependencies()
+    for each in dp.optional_dependencies():
+      if ambari.cluster_has_service(each.service_name):
+        merged_dependencies.add(each)
+    role_names = map(lambda each: each.service_name, merged_dependencies)
+    return AmbariTopologyUtil(ambari, role_names)
+  
+  @staticmethod
+  def get_knox_details(ambari):
+    return Knox(
+      user.url_input('Knox URL that is network accessible from DataPlane', 
+      'knox.url', 
+      default=str(ambari.cluster.knox_url())),
+      knox_user=ambari.cluster.knox_user(),
+      knox_group=ambari.cluster.knox_group())
+  
+  @staticmethod
+  def get_topologies_to_deploy(dp, ambari, topology_util):
+    return [
+        TokenTopology(dp.public_key()),
+        DpProxyTopology(ambari, dp.dependency_names(), 
+        topology_util)]
+  
+  @staticmethod
+  def check_cluster_for_dp(dp, knox):
+    return dp.check_ambari(knox)
+  
+  @staticmethod
+  def register_cluster_with_dp(dp, ambari, knox, user):
+    return dp.register_ambari(ambari, knox, user)
 
-  knox = Knox(user.url_input('Knox URL that is network accessible from DataPlane', 'knox.url', default=str(ambari.cluster.knox_url())), knox_user=ambari.cluster.knox_user(), knox_group=ambari.cluster.knox_group())
+"""
+  CMOperationsManager : control Operation for Clouder Manager based clusters
 
-  topologies_to_deploy = [TokenTopology(dp.public_key()), DpProxyTopology(ambari, dp.dependency_names(), topology_util)]
+"""
+class CMOperationsManager(BaseOperationsManager):
+  def __init__(self):
+    pass
+  @staticmethod
+  def get_instance():
+    return ClouderaManager(user.url_input('CM URL', 'cm.url'), user.credential_input('CM admin', 'cm.admin'))
```

Comment:
Shouldn't we have a similar print here like "Tell me about your Cloudera Manager instance" like we do for Ambari?